show/main.py

```diff
@@ -129,6 +176,150 @@ def run_command(command, display_cmd=False):
     if rc != 0:
         sys.exit(rc)
 
+
+def get_interface_mode():
+    mode = os.getenv('SONIC_CLI_IFACE_MODE')
```

Comment:
This solution is currently not working because you are using `os.getenv()` to check the `SONIC_CLI_IFACE_MODE` variable. However, this variable is not available to subprocesses (i.e., the "show" or "config" commands) because it is not exported in .bashrc.

You will need to modify the "config" command and add the `export` keyword when writing the `SONIC_CLI_IFACE_MODE` variable to the .bashrc file.

Reply:
SONIC_CLI_IFACE_MODE env variable is working fine acorss sessions/users. This is achieved by exporting the varaiable in  /etc/environment and sudoers file of the switch. 

Already raised a pull request  on this: 
https://github.com/Azure/sonic-buildimage/pull/1751/

We store variable in .bashrc (each user) for persistence. We finalised sudoers/environment approach because there is no direct way to change the  the variable of parent from the forked child bash instance. This already we agreed upon and the pull-request is raised on that design.

Reply:
I was under the assumption we abandoned the sudoers/environment approach (Azure/sonic-buildimage#1751) in favor of asking the user to log out and log back in again.

Reply:
We can safely discard that pull request. SONIC_CLI_IFACE_MODE variable changes are applied to bash environment when user logout and logback in to the new shell. So, there is no any necessity for /etc/environment and /etc/sudoers file to change the values in realtime. I raised a new pull-request in below with the above changes:

https://github.com/Azure/sonic-utilities/pull/321

Reply:
👍